core/ir/x86/decode.c

```diff
@@ -2579,6 +2580,16 @@ check_is_variable_size(opnd_t op)
 static byte *
 decode_common(dcontext_t *dcontext, byte *pc, byte *orig_pc, instr_t *instr)
 {
+    /*
+     * If we're dealing with decoding from synthetic ISA, we don't care about returning
+     * the pc of the next instruction (?), so we just write the encoding in final_pc and
+     * return it.
```

Comment:
?  We do need to return the next PC as that's the interface here; that is part of how failure is indicated as well, and the size of the instruction is determined.  Plus there might be multiple of these encoded back to back in some future use.

Reply:
Is it safe to assume that: next PC is simply current PC + instruction length for synthetic ISA?
Or am I missing some details?
I was looking at x86 encoding/decoding, but it's a but complicated, and I couldn't exactly convince myself.

Reply:
Yes if you know the length and it's a valid instruction (I think on invalid null is returned).

Reply:
Ok.